src/applyMiddleware.ts

```diff
@@ -51,7 +51,7 @@ export default function applyMiddleware<Ext, S = any>(
 ): StoreEnhancer<{ dispatch: Ext }>
 export default function applyMiddleware(
   ...middlewares: Middleware[]
-): StoreEnhancer {
+): StoreEnhancer<any> {
```

Comment:
this fix ends up being fine because the overloads provide more restrictive typing (you can check in `tests/typescript/middleware.ts` to see that the stores returned are strictly typed)